Dockerfile

```diff
@@ -1,160 +1,186 @@
+# As far as I can tell, this needs to be done in two steps.
+COPY --chown=0:0 static/  ./static/
+COPY --chown=0:0 *.json *.js  .
+RUN set -ex\
+	; npm install --quiet --no-progress --ignore-engines\
+	; npm run --quiet build\
+	;
+
+# Jwtproxy grabs jwtproxy.
+FROM stream as jwtproxy
+ENV OS=linux ARCH=amd64 
+ARG JWTPROXY_VERSION=0.0.3
+RUN set -ex\
+	; curl -fsSL -o /usr/local/bin/jwtproxy "https://github.com/coreos/jwtproxy/releases/download/v${JWTPROXY_VERSION}/jwtproxy-${OS}-${ARCH}"\
+	; chmod +x /usr/local/bin/jwtproxy\
+	;
+
+# Pushgateway grabs pushgateway.
+FROM stream AS pushgateway
+ENV OS=linux ARCH=amd64 
+ARG PUSHGATEWAY_VERSION=1.0.0
+RUN set -ex\
+	; curl -fsSL "https://github.com/prometheus/pushgateway/releases/download/v${PUSHGATEWAY_VERSION}/pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}.tar.gz"\
+	| tar xz "pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}/pushgateway"\
+	; install "pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}/pushgateway" /usr/local/bin/pushgateway\
+	;
+
+# Config-tool builds the go binary in the configtool.
+FROM docker.io/library/golang:1.15 as config-tool
 WORKDIR /go/src/config-tool
-RUN git clone https://github.com/quay/config-tool.git /go/src/config-tool
-RUN rm -rf /go/src/config-tool/pkg/lib/editor/static/build
-COPY --from=config-editor /config-editor/static/build  /go/src/config-tool/pkg/lib/editor/static/build
-
+ARG CONFIGTOOL_VERSION=master
+RUN curl -fsSL "https://github.com/quay/config-tool/archive/${CONFIGTOOL_VERSION}.tar.gz"\
+	| tar xz --strip-components=1 --exclude '*/pkg/lib/editor/static/build'
+COPY --from=config-editor /build/static/build  /go/src/config-tool/pkg/lib/editor/static/build
 RUN go install ./cmd/config-tool
 
-
-###################
-FROM registry.centos.org/centos/centos:8
-
+# Final is the end container, where all the work from the other
+# containers are copied in.
+FROM base AS final
 LABEL maintainer "thomasmckay@redhat.com"
 
-ENV OS=linux \
-    ARCH=amd64 \
-    PYTHON_VERSION=3.6 \
-    PATH=$HOME/.local/bin/:$PATH \
-    PYTHONUNBUFFERED=1 \
-    PYTHONIOENCODING=UTF-8 \
-    LC_ALL=C.UTF-8 \
-    LANG=C.UTF-8
-
-ENV QUAYDIR /quay-registry
-ENV QUAYCONF /quay-registry/conf
-ENV QUAYPATH "."
-
-# Switch to CentOS Stream
-RUN dnf install centos-release-stream -y
-RUN dnf swap centos-{linux,stream}-repos -y
-RUN dnf distro-sync -y
-RUN dnf upgrade -y
+# All of these chgrp+chmod commands are an Openshift-ism.
+#
+# Openshift runs a container as a random UID and GID 0, so anything
+# that's in the base image and needs to be modified at runtime needs
+# to make sure it's group-writable.
+RUN set -ex\
+	; setperms() { for d in "$@"; do chgrp -R 0 "$d" && chmod -R g=u "$d"; done; }\
+	; newdir() { for d in "$@"; do mkdir -m g+w "$d" || { chgrp 0 "$d" && chmod g=u "$d"; }; done; }\
+# Allow TLS certs to be created and installed as non-root user.
+# See also update-ca-trust(8).
+	; setperms /etc/pki/ca-trust/extracted /etc/pki/ca-trust/source/anchors\
```

Comment:
`conf/init/create_certs.sh` does not have the correct permission to copy its file in that directory.